packages/blocks/src/page-block/default/default-page-block.ts

```diff
@@ -193,6 +200,60 @@ function EmbedOptionContainer(embedOption: EmbedOption | null) {
     return html``;
   }
 }
+
+function handleUP(selection: DefaultSelectionManager) {
+  const { state } = selection;
+  if (state.selectedDividers.length === 1) {
+    const model = state.selectedDividers[0].model;
+    const container = getContainerByModel(model);
+    const preNodeModel = getPreviousBlock(container, model.id);
+    if (!preNodeModel || preNodeModel.id == '1') {
+      return;
+    } else if (
+      preNodeModel.flavour === 'affine:list' ||
+      preNodeModel.flavour === 'affine:paragraph'
+    ) {
+      focusPreviousBlock(model, 'end');
+      state.clear();
+      return;
+    } else if (preNodeModel.flavour === 'affine:divider') {
+      const selectionManager = getDefaultPageBlock(model).selection;
+      const dividerBlockElement = getBlockElementByModel(
+        preNodeModel
+      ) as HTMLElement;
+      const selectionRect = dividerBlockElement.getBoundingClientRect();
+      selectionManager.selectBlockByRect(selectionRect, preNodeModel);
+      state.type = 'divider';
+      return;
+    }
+  }
+}
+function handleDOWN(selection: DefaultSelectionManager) {
+  const { state } = selection;
+  if (state.selectedDividers.length === 1) {
+    const model = state.selectedDividers[0].model;
+    const nextBlock = getNextBlock(model.id);
+    if (!nextBlock) {
+      return;
+    } else if (
+      nextBlock.flavour === 'affine:list' ||
+      nextBlock.flavour === 'affine:paragraph'
+    ) {
+      focusNextBlock(model, 'start');
+      state.clear();
+      return;
+    } else if (nextBlock.flavour === 'affine:divider') {
```

Comment:
Please use `matchFlavours(['affine:divider'])` in *ALL* `block.flavour === 'affine:foo'` cases. This can make us able to better track *where* do we depend on such typing.